phoenix-core/src/main/java/org/apache/phoenix/coprocessor/GroupedAggregateRegionObserver.java

```diff
@@ -423,7 +426,14 @@ private RegionScanner scanUnordered(ObserverContext<RegionCoprocessorEnvironment
                         }
                     } while (hasMore && groupByCache.size() < limit);
                 }
-            } finally {
+            }  catch(NotServingRegionException e){
+                if(ScanUtil.isLocalIndex(scan)) {
+                    Exception cause = new StaleRegionBoundaryCacheException(c.getEnvironment().getRegion().getRegionInfo().getTable().getNameAsString());
+                    throw new DoNotRetryIOException(cause.getMessage(), cause);
+                } else {
+                    throw e;
```

Comment:
Is the throw ok because the client scanner knows how to deal with this in cases other than local indexes? Let's doc that, please.